### PR TITLE
Add Deposit Account Feature Spec

### PR DESCRIPTION
**Pull request by mcccoleman**
As part of the changes associated with [this](https://linear.app/retirable/issue/RET-2541/add-the-add-account-owners-endpoint-to-the-unit-ruby-gem) ticket, I'm taking the opportunity to flush out some of the specs associated with adding a deposit account. This ticket adds a new feature spec that confirms that we're creating and receiving information back from Unit when a deposit account is added.

### CODE REVIEW

File: spec/features/deposit_account_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+RSpec.describe Unit::DepositAccount do
+  before do
+    establish_connection_to_api!
+  end
+
+  let(:customer) do
+    Factory.create_individual_customer
+  end
+
+  let(:account) do
```
Comment (cisacke) [inline spec/features/deposit_account_spec.rb:13]: does this create a new customer & account every time `account` is referenced below? I'm seeing 100s of deposit accounts in the Unit sandbox environment, and can't tell if that's just every time the spec is run, or if multiple customers & accounts are being created per spec 🤔 it _might_ be a little cleaner if we did something similar to the ACH payment spec where the create is happening inside of the assertion?
i.e.
```Ruby
it 'creates a deposit account for the customer' do
  account = Unit::DepositAccount.create(deposit_product: 'checking', customer: customer)
  
  expect(account.id).to_not be nil
  etc etc
end
```

**Review by cisacke — APPROVED** [review]
one small question / suggestion but love the initiative here!

File: spec/features/deposit_account_spec.rb (hunk 1 above)
Comment (mcccoleman) [reply spec/features/deposit_account_spec.rb]: That's correct & good catch - I went ahead & updated to fix.

File: lib/unit-ruby/deposit_account.rb [hunk 2]
```diff
@@ -19,6 +19,18 @@ class DepositAccount < APIResource
     attribute :close_reason, Types::String, readonly: true # Optional. The reason the account was closed, either ByCustomer or Fraud.
 
     belongs_to :customer, class_name: 'Unit::IndividualCustomer'
+    has_many :customers, class_name: 'Unit::IndividualCustomer'
+
+    def add_customer(customer_id)
+      Unit::Connection.new.post(
```
Comment (ianyamey) [inline lib/unit-ruby/deposit_account.rb:25]:
```suggestion
      self.connection.post(
```

**Review by ianyamey — APPROVED** [review]

**Pull request merged**